cmd/object-handlers.go

```diff
@@ -2558,10 +2607,16 @@ func (api objectAPIHandlers) GetObjectRetentionHandler(w http.ResponseWriter, r
 		return
 	}
 
-	if _, err = getObjectInfo(ctx, bucket, object, opts); err != nil {
+	objInfo, err := getObjectInfo(ctx, bucket, object, opts)
```

Comment:
We should first check if the requester has the permissions to perform a GetObjectRetention request and only then fetch object info from the object layer.

Reply:
done